app/controllers/benefits/CompanyBenefitController.scala

```diff
@@ -74,7 +74,8 @@ class CompanyBenefitController @Inject()(
           }
 
           val form = {
-            val decision = currentCache.get(DecisionChoice)
+            val benefitType = currentCache.get(EndCompanyBenefit_BenefitTypeKey)
```

Comment:
When is this data set?

Reply:
It isn't, and doesn't need to be for the first visit, it is only set on the first submission of the form with these keys.